File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
        for(int i = 0;i<s.size();i++){
            if(!isalnum(s[i])){
                s.erase(s.begin()+i);
                i--;
            }
        }
        cout<<s;
        int l = 0, r = s.size()-1;
        while(l<=r){
            s[l]=tolower(s[l]);
            s[r]=tolower(s[r]);
            if(s[l]!=s[r])return false;
            l++;
            r--;
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int l = 0, r = s.length() - 1;
        String s1 = s.toLowerCase(); 
        while (l < r) {
            while (l < s1.length() && !Character.isLetterOrDigit(s1.charAt(l))) {
                l++;
            }
            while (r >= 0 && !Character.isLetterOrDigit(s1.charAt(r))) {
                r--;
            }
            if (l < r) {
                if (s1.charAt(l) != s1.charAt(r)) {
                    return false;
                }
                l++;
                r--;
            }
        }
        return true;
    }
}
